swap/src/bin/swap.rs

```diff
@@ -629,10 +692,43 @@ where
     Ok((btc_swap_amount, fees))
 }
 
+pub async fn check_latest_version() -> Result<()> {
+    const GITHUB_LATEST_VERSION_URL: &str =
+        "https://github.com/comit-network/xmr-btc-swap/releases/latest";
+
+    let response = match reqwest::get(GITHUB_LATEST_VERSION_URL).await {
+        Ok(res) => res,
+        Err(_) => bail!(
```

Comment:
I don't think it should ever bail if it can't check for an update.

Reply:
should I catch and print the exception inside the function instead that calling?

Reply:
oh sorry! you are actually matching on the error and logging at the callsite so I think that's fine 👍